controllers/database.go

```diff
@@ -28,6 +28,7 @@ var dbTemplates = []string{
 	"mariadb/pvc.yaml.tmpl",
 	"mariadb/sa.yaml.tmpl",
```

Comment:
what's the reasoning we don't use this SA anyways?

Reply:
You're right, the `sa.yaml.tmpl` isn't being used anywhere. I think the idea suggested by the issue: https://github.com/opendatahub-io/data-science-pipelines-operator/issues/204 to have two SAs, one default SA and one non-default. Issue https://github.com/opendatahub-io/data-science-pipelines-operator/issues/191 further mentions an error encountered in connecting the DB and the fix for it was creating another SA. 

The issue here could be that the original `sa.yaml.tmpl` SA wasn't being referenced in the mariaDB deployment. @HumairAK wdyt?

Reply:
> https://github.com/opendatahub-io/data-science-pipelines-operator/issues/204 to have two SAs, one default SA and one non-default.

the default SA being referred to in #204 is the SA named `default` that exists in every namespace. By specifying a specific SA to use, we are already using the non-default SA. I didn't know we had an SA already for mariadb not being used, if that's the case let's use this one for mariadb

Reply:
Done, updated to have just one SA for mariaDB, and updated the references in the mariaDB deployment and the controller code to point to this SA.